File: Python/Ejercicio 1/Interfaz_Consola.py
from Gestion_de_Vocabulario import agregar_palabra, buscar_palabra, editar_palabra, eliminar_palabra
from Modo_de_Evaluacion import iniciar_evaluacion

def menu():
    """Menu principal en la consola de usuario"""
    while True:
        print("\n--- Menú Principal ---")
        print("1. Agregar nueva palabra")
        print("2. Buscar palabra")
        print("3. Editar palabra")
        print("4. Eliminar palabra")
        print("5. Iniciar evaluación")
        print("6. Salir")

        opcion = input("Seleccione una opción: ")

        if opcion == "1":
            palabra = input("Ingrese la palabra en ingles: ")
            significado = input("Ingrese el significado en español: ")
            agregar_palabra(palabra, significado)

        elif opcion == "2":
            palabra = input("Ingrese la palabra a buscar: ")
            buscar_palabra(palabra)

        elif opcion == "3":
            palabra = input("Editar una palabra: ")
            nueva_palabra = input("Nueva palabra en ingles: ")
            nuevo_significado = input("Nuevo significado de la palabra: ")
            editar_palabra(palabra, nueva_palabra, nuevo_significado)

        elif opcion == "4":
            palabra = input("Eliminar una palabra: ")
            eliminar_palabra(palabra)

        elif opcion == "5":
            iniciar_evaluacion()

        elif opcion == "6":
            print("Sistema finalizado. hasta luego.")
            break

        else:
            print("Opción no válida. Intentelo de nuevo.")

if __name__ == "__main__":
    menu()

File: Typescript/Ejercicio 1/interfazConsola.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var readline_sync_1 = require("readline-sync");
var gestionVocabulario_1 = require("./gestionVocabulario");
var modoEvaluacion_1 = require("./modoEvaluacion");
function menu() {
    while (true) {
        console.log("\n--- Menú Principal ---");
        console.log("1. Agregar palabra");
        console.log("2. Buscar palabra");
        console.log("3. Editar palabra");
        console.log("4. Eliminar palabra");
        console.log("5. Iniciar evaluación");
        console.log("6. Salir");
        var op = (0, readline_sync_1.question)("Seleccione una opcion: ");
        switch (op) {
            case "1":
                var p = (0, readline_sync_1.question)("Palabra en ingles: ");
                var s = (0, readline_sync_1.question)("Significado: ");
                (0, gestionVocabulario_1.agregarPalabra)(p, s);
                break;
            case "2":
                (0, gestionVocabulario_1.buscarPalabra)((0, readline_sync_1.question)("Buscar: "));
                break;
            case "3":
                var pe = (0, readline_sync_1.question)("Editar palabra: ");
                var np = (0, readline_sync_1.question)("Nueva palabra: ");
                var ns = (0, readline_sync_1.question)("Nuevo significado: ");
                (0, gestionVocabulario_1.editarPalabra)(pe, np, ns);
                break;
            case "4":
                (0, gestionVocabulario_1.eliminarPalabra)((0, readline_sync_1.question)("Eliminar: "));
                break;
            case "5":
                (0, modoEvaluacion_1.iniciarEvaluacion)();
                break;
            case "6":
                console.log("Adios!");
                return;
            default:
                console.log("Opcion invalida.");
        }
    }
}
menu();
